Gems/Multiplayer/Code/Source/Components/NetworkTransformComponent.cpp

```diff
@@ -82,15 +87,33 @@ namespace Multiplayer
 
     void NetworkTransformComponent::OnResetCountChangedEvent()
     {
+        OnParentIdChangedEvent(GetParentEntityId());
+
         m_targetTransform.SetRotation(GetRotation());
         m_targetTransform.SetTranslation(GetTranslation());
         m_targetTransform.SetUniformScale(GetScale());
         m_previousTransform = m_targetTransform;
     }
 
+    void NetworkTransformComponent::OnParentIdChangedEvent([[maybe_unused]] NetEntityId newParent)
```

Comment:
Oh, you actually hooked this up! interesting.. I'm really curious how do these two systems interact, the network(root/child)hierarchy components and this transform parenting concept?

Reply:
Yep! I had to hook this up, otherwise, parenting changes on a server had no effect on clients. I have a unit test for this actually already in this CR, where I push a new NetEntityId as a parent, have NetworkTransformComponent push that onto TransformComponent, which is then picked up by Hierarchy components that rebuild their hierarchies as necessary.

Reply:
Unittests for this are: Client_Sends_NetworkHierarchy_Updated_Event_On_Child_Detached_On_Server and Client_Sends_NetworkHierarchy_Leave_Event_On_Child_Detached_On_Server.